gcloud/pubsub/topic.py

```diff
@@ -113,6 +122,8 @@ def publish(self, message, **attrs):
         :rtype: str
         :returns: message ID assigned by the server to the published message
         """
+        if self.add_timestamp_to_messages:
+            attrs['timestamp'] = '%sZ' % _NOW().isoformat()
```

Comment:
OK. I'm still torn, but can't find a way to come down on either side, so let's move forward.